### ISSUE TITLE
how to prevent XSS?

### ISSUE THREAD

**Issue by Caratpine** [softvar/json2html#27]
thanks for your project.
it's pretty good, I have a question: how to prevent XSS?
does this project provide this function?
thanks :-)

**Comment by muellermichel** [comment]
Hey Caratpine. If you use my [fork](https://github.com/muellermichel/json2html) (or if I could get softvar to finally merge my [PR](https://github.com/softvar/json2html/pull/24) you get XSS protection included. For that you simply pass `encode=True` to the convert call, this will escape everything nasty and also guarantee that the html looks good for non-ascii characters. Alternatively you can call `.encode('ascii', 'xmlcharrefreplace')` on the string returned by convert (and you should do that for any other untrusted data that you return back by http response).

**Comment by softvar** [comment]
@muellermichel is right. Also, #24 is now merged. So, the latest version 1.2.0 will fix this.

**Comment by Caratpine** [comment]
@muellermichel  @softvar thanks for your relies😄

I use this example:
```
from json2html import json2html
from flask import Flask
import json


app = Flask(__name__)


@app.route('/')
def index():
    s = "<script></script>"
    return json2html.convert(json=json.dumps(s), encode=True)


app.test_client().get('/').data

```

it returns `<script></script>` not ` '&lt;script&gt;&lt;/script&gt;'`


so i use this method to prevent XSS:

```
def escape(s):
    return s.replace('&', '&amp;').replace('>', '&gt;').replace('<', '&lt;')


s = '<script></script>
s = json2html.convert(json=escape(s))

```

**Comment by muellermichel** [comment]
Ah I see your usecase - sorry I was wrong, `xmlcharrefreplace` in that case doesn't do what you want, it only escapes non-html characters to html. I think it would make sense to do your escaping on the text node level by default. For that we could simply use `cgi.escape` for all text node types. Doing so by default would be analogous to what most templating engines do.

Btw. `cgi.escape` does exactly what your `escape` function is doing - more info here: https://wiki.python.org/moin/EscapingHtml

I therefore advocate to leave this issue open. The code to do this with json2html is only 3 lines plus some additional tests, so it wouldn't be a big deal to support.

**Comment by Caratpine** [comment]
@muellermichel  thanks for your patience to answer 😄

**Comment by muellermichel** [comment]
I've created the PR linked above. See lines 191 and follwing in https://github.com/muellermichel/json2html/blob/master/test/run_tests.py : This should fix XSS for good. 

However there is one drawback: There is currently no whitelisting going on. If someone uses json2html together with text nodes containing formatting tags (like <b></b> as an example), it would escape it and break formatting, so in that case the user would now have to set escape=False. Since formatting was never a documented feature in json2html I think this should be OK however.

**Comment by softvar** [comment]
Thanks for pointing this out @Caratpine 
It's fixed now in version 1.2.1.

Thanks @muellermichel for the patch.
